util/httputil/client.go

```diff
@@ -125,6 +127,21 @@ type TLSOptions struct {
 	ServerName         string
 }
 
+func TLSConfig(cfg config.TLSConfig) (*tls.Config, error) {
```

Comment:
> This should all move to using the tlsconfig over in common, as this is not the only place in this organisation we need to deal with tls configuration.

I agree, shall we do baby steps?  A PR which gets rid of the intermediate form in httputil?